code/modules/mob/new_player/poll.dm

```diff
@@ -3,480 +3,322 @@
+	if(!query_insert.Execute())
+		var/err = query_insert.ErrorMsg()
+		log_game("SQL ERROR adding text reply to table. Error : \[[err]\]\n")
+		return
+	usr << "<span class='notice'>Feedback logging successful.</span>"
+	usr << browse(null,"window=playerpoll")
 
-		var/DBQuery/insert_query = dbcon.NewQuery("INSERT INTO [format_table_name("poll_vote")] (id ,datetime ,pollid ,optionid ,ckey ,ip ,adminrank, rating) VALUES (null, Now(), [pollid], [optionid], '[usr.ckey]', '[usr.client.address]', '[adminrank]', [(isnull(rating)) ? "null" : rating])")
-		insert_query.Execute()
+/mob/new_player/proc/vote_on_numval_poll(pollid, optionid, rating)
+	if(!pollid || !optionid || !rating)
+		return
+	var/adminrank = poll_check_voted(pollid, "poll_vote")
+	if(!adminrank)
+		return
+	var/DBQuery/query_insert = dbcon.NewQuery("INSERT INTO [format_table_name("poll_vote")] (datetime ,pollid ,optionid ,ckey ,ip ,adminrank, rating) VALUES (Now(), [pollid], [optionid], '[ckey]', '[client.address]', '[adminrank]', [(isnull(rating)) ? "null" : rating])")
+	if(!query_insert.Execute())
+		var/err = query_insert.ErrorMsg()
+		log_game("SQL ERROR adding vote to table. Error : \[[err]\]\n")
+		return
+	usr << "<span class='notice'>Vote successful.</span>"
+	usr << browse(null,"window=playerpoll")
 
-		usr << "<span class='notice'>Vote successful.</span>"
-		usr << browse(null,"window=playerpoll")
+/mob/new_player/proc/vote_on_multi_poll(pollid, optionid)
+	if(!pollid || !optionid)
+		return
+	if(!dbcon.IsConnected())
+		usr << "<span class='danger'>Failed to establish database connection.</span>"
+		return
+	var/DBQuery/query_get_choicelen = dbcon.NewQuery("SELECT multiplechoiceoptions FROM [format_table_name("poll_question")] WHERE pollid = [pollid]")
+	if(!query_get_choicelen.Execute())
+		var/err = query_get_choicelen.ErrorMsg()
+		log_game("SQL ERROR obtaining multiplechoiceoptions from poll_question table. Error : \[[err]\]\n")
+		return
+	var/i = text2num(query_get_choicelen.item[1])
+	var/DBQuery/query_hasvoted = dbcon.NewQuery("SELECT id FROM [format_table_name("poll_vote")] WHERE pollid = [pollid] AND ckey = '[ckey]'")
+	if(!query_hasvoted.Execute())
+		var/err = query_hasvoted.ErrorMsg()
+		log_game("SQL ERROR obtaining id from poll_vote table. Error : \[[err]\]\n")
+		return
+	while(i)
+		if(query_hasvoted.NextRow())
+			i--
+	if(!i)
+		usr << "<span class='danger'>Maximum replies reached.</span>"
+		return
+	var/adminrank = "Player"
+	if(client.holder)
+		adminrank = client.holder.rank
+	var/DBQuery/query_insert = dbcon.NewQuery("INSERT INTO [format_table_name("poll_vote")] (datetime, pollid, optionid, ckey, ip, adminrank) VALUES (Now(), [pollid], [optionid], '[ckey]', '[client.address]', '[adminrank]')")
+	if(!query_insert.Execute())
+		var/err = query_insert.ErrorMsg()
+		log_game("SQL ERROR adding vote to table. Error : \[[err]\]\n")
+		return
+	usr << "<span class='notice'>Vote successful.</span>"
+	usr << browse(null,"window=playerpoll")
```

Comment:
These should probably be an `if...else`.  Also, the player should be notified that the vote failed to take.